xdsl/irdl.py

```diff
@@ -947,7 +947,7 @@ def opt_successor_def(
     return cast(OptSuccessor, _SuccessorFieldDef(OptSuccessorDef))
 
 
-_OPERATION_DICT_KEYS = set(key for cls in Operation.mro() for key in cls.__dict__)
+_OPERATION_DICT_KEYS = set(key for cls in Operation.mro()[:-1] for key in cls.__dict__)
```

Comment:
What does the `[:-1]` remove?
Also, should this be `IRDLOperation` instead?

Reply:
`object`, IRDLOperation doesn't define new properties, and I don't think we're planning to add new properties in it? In any case  the tests will go boom if we do, so we should catch the error at that point.

Reply:
Right, as long as tests will catch it, that's good!